src/colinajose/service/SchoolService.java

```diff
@@ -22,188 +23,265 @@
         MyCircularDoublyLinkedList<Grade> sortedGrades = GradesService.sortGrades(grades);
         MyArrayList<MyHashMap<String, String>> entriesToWrite = gradesToEntries(sortedGrades, studentField, gradeField);
         return dataHandler.write(pathFile, entriesToWrite);
     }
-    private MyArrayList<MyHashMap<String, String>> gradesToEntries(MyCircularDoublyLinkedList<Grade> grades, String studentField, String gradeField){
+
+    private MyArrayList<MyHashMap<String, String>> gradesToEntries(MyCircularDoublyLinkedList<Grade> grades, String studentField, String gradeField) {
         MyArrayList<MyHashMap<String, String>> entries = new MyArrayList<>();
-        for (int i = 0; i < grades.size(); i++) {
-            MyHashMap<String, String> entry = new MyHashMap<>();
-            entry.put(studentField, grades.get(i).getStudent().getName());
-            entry.put(gradeField, String.valueOf(grades.get(i).getFinalGrade()));
-            entries.add(entry);
+        if (grades.size() != 0) {
+            for (int i = 0; i < grades.size(); i++) {
+                MyHashMap<String, String> entry = new MyHashMap<>();
+                entry.put(studentField, grades.get(i).getStudent().getName());
+                entry.put(gradeField, String.valueOf(grades.get(i).getFinalGrade()));
+                entries.add(entry);
+            }
         }
         return entries;
     }
-    public MyCircularDoublyLinkedList<Student> getStudents(String kardexId){
+
+    public MyCircularDoublyLinkedList<Student> getStudents(String kardexId) {
         Kardex kardex = SearchService.getKardex(this.school, kardexId);
         return kardex.getStudents();
     }
-    public MyCircularDoublyLinkedList<Subject> getSubjects(String courseId){
-        Course course = SearchService.getCourse(this.school, courseId);
-        return course.getSubjects();
-    }
-    public MyCircularDoublyLinkedList<Grade> getGrades(String kardexId){
+
+    public MyCircularDoublyLinkedList<Grade> getGrades(String kardexId) {
         Kardex kardex = SearchService.getKardex(this.school, kardexId);
         return GradesService.computeGrades(kardex);
     }
-    public Teacher getSubjectTeacher(String subjectId, String courseId){
+
+    public Teacher getSubjectTeacher(String subjectId, String courseId) {
         Course course = SearchService.getCourse(this.school, courseId);
         Subject subject = SearchService.getSubject(course, subjectId);
         return subject.getTeacher();
     }
-    public Teacher getTeacher(String teacherId){
+
+    public Teacher getTeacher(String teacherId) {
         return SearchService.getTeacher(this.school, teacherId);
     }
-    public Student getStudent(String studentId){
+
+    public Student getStudent(String studentId) {
         return SearchService.getStudent(this.school, studentId);
     }
+
+    private <T extends BaseEntity> MyArrayList<MyHashMap<String, String>> elementsToEntries(MyCircularDoublyLinkedList<T> elements) {
```

Comment:
Jose, I don't like this method you are not utilizing Map correctly. This method should return a Map not a list of Maps.
The returned Map keys will be the Ids and the values will be the actual object (T).
As this is you are tightly coupling the Services with the Presentation. Handling everything as a String is not ok.
If need to, please modify the MyHashMap class to include two methods.
public MyArrayList<T> getKeys();
public MyArrayList<U> getValues();

Reply:
@JurgenPadilla fixed in next commit